File: algorithms/fib.py
def fib(n):
    if n < 2:
        return n
    return fib(n-1) + fib(n - 2)


def fib_opt(n, cache={}):
    if n < 2:
        return n
    elif n in cache:
        return cache[n]
    else:
        cache[n] = fib_opt(n - 1, cache) + fib_opt(n - 2, cache)
        return cache[n]


print(fib_opt(996))

File: algorithms/fib.js
// Initialize an empty array of length n. Use reduce to add values to the array, using the sum of the last two values, except for the first two

const fib = (n) =>
	[...Array(n)].reduce((acc, val, i) => acc.concat(i > 1 ? acc[i - 2] : i), []);

console.log(fib(50));

const fibonacci = (n) => {
	if (n < 2) {
		return n;
	}

	return fibonacci(n - 1) + fibonacci(n - 2);
};

console.log(fibonacci(10));
